internal/repository/oci/ocirepo.go

```diff
@@ -0,0 +1,121 @@
+package oci
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	containerregistryv1 "github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/kyma-project/lifecycle-manager/internal/manifest/spec"
+)
+
+var (
+	ErrKeyChainNotNil   = errors.New("keychain lookup must not be nil")
+	ErrNoProtocolScheme = errors.New("hostref must not contain protocol scheme (http/https)")
+	ErrNoLeadingSlash   = errors.New("hostref must not start with a '/'")
+)
+
+// RepositoryReader provides basic support to read data from OCI repositories.
+type RepositoryReader struct {
+	keyChainLookup spec.KeyChainLookup
+	hostref        string
```

Comment:
Each part of URL have a standard name, this is just `host`, I never heard of `hostref`.